Rationale.md

```diff
@@ -156,22 +156,33 @@ TODO
 
 ## Indirect Calls
 
-The exact semantics of indirect function calls, function pointers, and what
-happens when calling the wrong function, are still being discussed.
-
-Fundamentally linear memory is a simple collection of bytes, which means that
-some integral representation of function pointers must exist. It's desirable to
-hide the actual address of generated code from untrusted code because that would
-be an unfortunate information leak which could have negative security
-implications. Indirection is therefore desired.
-
-One extra concern is that existing C++ code sometimes stores data inside of what
-is usually a function pointer. This is expected to keep working.
-
-Dynamic linking further complicates this: WebAssembly cannot simply standardize
-on fixed-size function tables since dynamically linked code can add new
-functions, as well as remove them.
-
+The table-based scheme for indirect function calls was motivated by the need
+to represent function pointers as integer values that can be stored into the
+linear memory, as well as to enforce basic safety properties such as a
```

Comment:
On Fri, Oct 23, 2015 at 11:56 AM, JF Bastien <notifications@github.com>
wrote:

> In Rationale.md
> <https://github.com/WebAssembly/design/pull/396#discussion_r42901416>:
>
> > -Fundamentally linear memory is a simple collection of bytes, which means that
> > -some integral representation of function pointers must exist. It's desirable to
> > -hide the actual address of generated code from untrusted code because that would
> > -be an unfortunate information leak which could have negative security
> > -implications. Indirection is therefore desired.
> > -
> > -One extra concern is that existing C++ code sometimes stores data inside of what
> > -is usually a function pointer. This is expected to keep working.
> > -
> > -Dynamic linking further complicates this: WebAssembly cannot simply standardize
> > -on fixed-size function tables since dynamically linked code can add new
> > -functions, as well as remove them.
> > -
> > +The table-based scheme for indirect function calls was motivated by the need
> > +to represent function pointers as integer values that can be stored into the
> > +linear memory, as well as to enforce basic safety properties such as a
>
> The extra "a" is still there?
>
Done.


> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WebAssembly/design/pull/396/files#r42901416>.
>